src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -778,6 +778,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+                tupleTypeOpt.CheckAllConstraints(this.Conversions, node.Location, diagnostics);
```

Comment:
Sorry, I had missed that comment. Aleksey clarified why `ShouldCheckConstraints` is only meaningful for type (but not expression).